File: Entity/ConsultaOdontologica.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Entity
{
    public class ConsultaOdontologica
    {
        [Key]
        public int IdConsultaOdontologica { get; set; }
        public string Motivo { get; set; }
        public string Medicacion { get; set; }
        public string Diagnostico { get; set; }
        public string Valoracion { get; set; }
        public string RecetaMedica { get; set; }

        public int IdAntecedentes { get; set; }
        public Antecedente Antecedente { get; set; }

        public string NoDocumento { get; set; }
        [JsonIgnore]
        public Paciente Paciente { get; set; }

        public int IdSolicitudCita { get; set; }
        [JsonIgnore]
        public SolicitudCita SolicitudCita { get; set; }

        public int IdServicio { get; set; }
        [JsonIgnore]
        public Servicio Servicio { get; set; }


    }
}

File: DentalTime/ClientApp/src/app/@elements/models/ConsultaOdontologica.ts
import { Antecendentes } from "./Antecendentes";
import { Paciente } from "./Paciente";
import { Servicio } from "./Servicio";
import { SolicitudCita } from "./SolicitudCita";

export class ConsultaOdontologica {

    motivo:String;
    medicacion: string;
    diagnostico : string;
    valoracion: string;
    recetaMedica: string;
    antecedente: Antecendentes;
    noDocumento:String;
    idSolicitudCita: Number;
    IdServicio: Number;

}

export class ConsultaOdontologicaView extends ConsultaOdontologica{

    codConsultaClinica:number;
    servicio:Servicio;
    solicitudCita:SolicitudCita;
    paciente: Paciente;

}
